back-end/API/README.md

```diff
@@ -45,24 +64,93 @@ flask run
 flask run --host=0.0.0.0
 ```
 
+### Routes
+#### /user
+- /createUser
+    - POST
+    - Créer un utilisateur
+    - Token requis
+    - Paramètres:
+        - email: string
+        - password: string
+
+- /login
+    - POST
+    - Se connecter
+    - Paramètres:
+        - email: string
+        - password: string
+
+- /updatePassword
+    - PUT
+    - Mettre à jour le mot de passe
+    - Token requis
+    - Paramètres:
+        - email: string
+        - password: string
+
+- /getUser
+    - GET
+    - Récupérer un utilisateur
+    - Token requis
+    - Paramètres:
+        - email: string
+
+- /getAllUsers
+    - GET
+    - Récupérer tous les utilisateurs
+    - Token requis
+
+#### /jobOffer
+
+- /offreEmploi/:id
+    - GET
+    - Récupérer une offre d'emploi selon l'id
+    - Paramètres:
+        - id: int
+
+- /offresEmploi
+    - GET
+    - Récupérer toutes les offres d'emploi
+
+
 ### Running the tests
 
 Créer la base de données de test avec ce script:
 ```sql
 CREATE DATABASE H2024test;
 USE H2024test;
 CREATE TABLE user (
-    id int not null auto_increment,
-    name varchar(255) ,
-    email varchar(255) ,
-    password varchar(255) ,
-    admin boolean ,
-    primary key (id)
+    id SERIAL PRIMARY KEY,
+    email VARCHAR(255) NOT NULL,
+    password VARCHAR(255) DEFAULT FALSE,
+    active BOOLEAN DEFAULT FALSE,
+    isModerator BOOLEAN DEFAULT FALSE
+);
+CREATE TABLE IF NOT EXISTS job_offer (
+    id INT AUTO_INCREMENT PRIMARY KEY,
+    title VARCHAR(255) NOT NULL,
+    address VARCHAR(255) NOT NULL,
+    description VARCHAR(255) NOT NULL,
+    dateEntryOffice DATETIME NOT NULL,
+    deadlineApply DATE NOT NULL,
+    email VARCHAR(255) NOT NULL,
+    hoursPerWeek FLOAT NOT NULL,
+    compliantEmploymentStandards BOOLEAN NOT NULL,
+    internship BOOLEAN NOT NULL,
+    offerStatus INT NOT NULL,
+    offerLink VARCHAR(255) NOT NULL,
+    urgent BOOLEAN NOT NULL,
+    active BOOLEAN NOT NULL,
+    employerId INT NOT NULL,
+    scheduleId INT NOT NULL
 );
+
 CREATE USER 'admin'@'localhost' IDENTIFIED BY 'admin';
 GRANT ALL PRIVILEGES ON H2024test.* TO 'root'@'localhost';
```

Comment:
cela ne devrait pas être plutôt : GRANT ALL PRIVILEGES ON H2024test.* TO 'admin'@'localhost'; au lieu de l'utilisateur "root"?